include/alpaka/core/BoostPredef.hpp

```diff
@@ -42,8 +42,9 @@
 
 // hip compiler detection
 #if !defined(BOOST_COMP_HIP)
-#    if defined(__HIP__)
-#        define BOOST_COMP_HIP BOOST_VERSION_NUMBER_AVAILABLE
+#    if defined(__HIP__) && defined(__HIPCC__) // We are using clang to compile HIP code
+#        include <hip/hip_version.h>
+#        define BOOST_COMP_HIP BOOST_VERSION_NUMBER(HIP_VERSION_MAJOR, HIP_VERSION_MINOR, HIP_VERSION_PATCH)
```

Comment:
Just a note that `HIP_VERSION_PATCH` may not be what you think it is: in HIP/ROCm 5.4.3 they use
```c++
#define HIP_VERSION_MAJOR 5
#define HIP_VERSION_MINOR 4
#define HIP_VERSION_PATCH 22804
```
so the last number is a bit complicated to use :)

Reply:
Thanks for the warning!

Reply:
OK, turns out that `HIP_VERSION` is defined in the same way as `BOOST_COMP_HIP` (if I counted the digits correctly).

In fact, in `include/alpaka/core/ApiHipRt.hpp` I use `HIP_VERSION` instead of relying on one of the others, because I need to check what version of HIP is available while compiling host code with a host compiler (e.g. `gcc`).